### PR TITLE
Replace String on Into<String> for all TestDir methods (#387)

### LINKED ISSUE #387: Replace String on Into<String> for all TestDir methods
### Description

We use the `TestDir` struct in our integrational tests. Some methods of this struct take `String` as an argument:
* [test_command_success](https://github.com/dotenv-linter/dotenv-linter/blob/master/tests/common/test_dir.rs#L71)
* [test_command_fail](https://github.com/dotenv-linter/dotenv-linter/blob/master/tests/common/test_dir.rs#L79)
* [test_command_success_with_args](https://github.com/dotenv-linter/dotenv-linter/blob/master/tests/common/test_dir.rs#L88)
* [test_command_fail_with_args](https://github.com/dotenv-linter/dotenv-linter/blob/master/tests/common/test_dir.rs#L108)
* [test_command_fix_success](https://github.com/dotenv-linter/dotenv-linter/blob/master/tests/common/test_dir.rs#L128)
* [test_command_fix_success_with_args](https://github.com/dotenv-linter/dotenv-linter/blob/master/tests/common/test_dir.rs#L154)

In some tests we convert `&str` to string using the `String::from` method:

```rust
// tests/output/check.rs
#[test]
fn problems() {
    let test_dir = TestDir::new();
    test_dir.create_testfile(".env", "abc=DEF\n");

    let expected_output = String::from(
        r#"Checking .env
.env:1 LowercaseKey: The abc key should be in uppercase

Found 1 problem
"#,
    );

    test_dir.test_command_fail(expected_output);
}
```

It would be great if these methods take `&str` as well.

### Implementation

We can do it using the `Into<String>` trait.

### PR DESCRIPTION
**Pull request by ebobrow**
Closes #387

#### ✔ Checklist:

- [x] Commit messages have been written in [Conventional Commits](https://www.conventionalcommits.org) format;
- [x] This PR has been added to [CHANGELOG.md](https://github.com/dotenv-linter/dotenv-linter/blob/master/CHANGELOG.md) (at the top of the list);

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add benchmark for the check function [#376](https://github.com/dotenv-linter/dotenv-linter/pull/376) ([@mgrachev](https://github.com/mgrachev))
 
 ### 🔧 Changed
+- Replace String on Into<String> for all TestDir methods [#387](https://github.com/dotenv-linter/dotenv-linter/pull/387) ([@ebobrow](https://github.com/ebobrow))
```
Comment (mgrachev) [inline CHANGELOG.md:13]:
```suggestion
- Replace String on Into<String> for all TestDir methods [#397](https://github.com/dotenv-linter/dotenv-linter/pull/397) ([@ebobrow](https://github.com/ebobrow))
```

**Review by mgrachev — COMMENTED** [review]
@ebobrow 👋 Thank you for your contribution 🚀

I have left a comment. Please take a look at it 👀

Also, please fix a warning from [commitlint](https://github.com/dotenv-linter/dotenv-linter/pull/397/checks?check_run_id=2157502837).

**Comment by ebobrow** [comment]
Thank you! Sorry, I'm new to open source.

**Comment by DDtKey** [comment]
@ebobrow thank you for your PR 👍 
Can you resolve conflict from the master branch?

**Review by DDtKey — APPROVED** [review]

**Review by mgrachev — APPROVED** [review]

**Comment by mgrachev** [comment]
@ebobrow Thank you for your help ❤️

🙏 If it’s not difficult for you, please support the project - star on GitHub ⭐️

**Pull request merged**